server/src/test/java/org/elasticsearch/repositories/fs/FsBlobStoreRepositoryIntegTests.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.fs;
+
+public final class FsBlobStoreRepositoryIntegTests extends FsBlobStoreRepositoryBaseIntegTestCase {
```

Comment:
I have made the `repositorySettings` abstract, rather than the `repositoryType`.
It felt bizarre to have the type abstract since the type is in the class name. It's better to have the settings abstract because settings can differ a lot between classes in the same  hierarchy.

Reply:
Let me know if this looks OK for you.

Reply:
> It felt bizarre to have the type abstract since the type is in the class name.

I would've renamed the class then maybe :) But let's not hold this PR up on that IMO. I think this change also makes things clear enough for now. Thanks!